src/app/taquin/taquin.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit, OnChanges, Input } from '@angular/core';
+import {shuffle} from 'lodash';
+
+@Component({
+  selector: 'app-taquin',
+  templateUrl: './taquin.component.html',
+  styleUrls: ['./taquin.component.css']
+})
+
+
+
+export class TaquinComponent implements OnInit, OnChanges {
+
+  @Input() TilesNumber: number;
+  Order: Array<number> = [];
+  Tilesstr = '';
+  Win = false;
+  counter = 0;
+  constructor() {}
+
+  ngOnInit(): void {
+    this.Tilesstr = this.TilesNumber.toString();
+    this.Order = shuffle([...Array(this.TilesNumber).keys()]);
+  }
+
+  ngOnChanges(): void {
+    this.Tilesstr = this.TilesNumber.toString();
+    this.Order = shuffle([...Array(this.TilesNumber).keys()]);
+  }
+
+  sourceContent(content: number): string {
+    const src: string = 'assets/' + this.Tilesstr + '/';
+    if (content !== 0) {
+      return src + String(content) + '.jpg';
+    } else {
+      return src + 'final.jpg';
+    }
+  }
+
+  CanSwap(index0: number, indexNum: number): boolean {
```

Comment:
Prefer `camelCase` for method names too
```suggestion
  canSwap(index0: number, indexNum: number): boolean {
```

Reply:
Good method name though since it returns a boolean 👏